File: ios/Classes/Models/HMSTokenResultExtension.swift
//
//  HMSTokenResultExtension.swift
//  hmssdk_flutter
//
//  Created by Pushpam on 27/03/23.
//

import Foundation

class HMSTokenResultExtension{
    
    static func toDictionary(_ hmsTokenResult: String?)  -> [String: String?] {
        var dict = [String: String?]()

        dict["auth_token"] = hmsTokenResult
        dict["expires_at"] =  nil

        return dict
    }
}

File: android/src/main/kotlin/live/hms/hmssdk_flutter/HMSTokenResultExtension.kt
package live.hms.hmssdk_flutter

import live.hms.video.signal.init.TokenResult

class HMSTokenResultExtension {

    companion object {
        fun toDictionary(hmsTokenResult: TokenResult):HashMap<String,String?>{
            val hashMap = HashMap<String, String?>()

            hashMap["auth_token"] = hmsTokenResult.token
            hashMap["expires_at"] = hmsTokenResult.expiresAt

            return hashMap
        }
    }
}